src/commands/client.ts

```diff
@@ -74,59 +74,33 @@ export class CommandClient extends Client implements CommandClientOptions {
   constructor(options: CommandClientOptions) {
     super(options)
     this.prefix = options.prefix
-    this.mentionPrefix =
-      options.mentionPrefix === undefined ? false : options.mentionPrefix
-
-    this.getGuildPrefix =
-      options.getGuildPrefix === undefined
-        ? (id: string) => this.prefix
-        : options.getGuildPrefix
-    this.getUserPrefix =
-      options.getUserPrefix === undefined
-        ? (id: string) => this.prefix
-        : options.getUserPrefix
-
-    this.getChannelPrefix =
-      options.getChannelPrefix === undefined
-        ? (id: string) => this.prefix
-        : options.getChannelPrefix
-
-    this.isUserBlacklisted =
-      options.isUserBlacklisted === undefined
-        ? (id: string) => false
-        : options.isUserBlacklisted
-    this.isGuildBlacklisted =
-      options.isGuildBlacklisted === undefined
-        ? (id: string) => false
-        : options.isGuildBlacklisted
-    this.isChannelBlacklisted =
-      options.isChannelBlacklisted === undefined
-        ? (id: string) => false
-        : options.isChannelBlacklisted
-
-    this.spacesAfterPrefix =
-      options.spacesAfterPrefix === undefined
-        ? false
-        : options.spacesAfterPrefix
-
-    this.owners = options.owners === undefined ? [] : options.owners
-    this.allowBots = options.allowBots === undefined ? false : options.allowBots
-    this.allowDMs = options.allowDMs === undefined ? true : options.allowDMs
-    this.caseSensitive =
-      options.caseSensitive === undefined ? false : options.caseSensitive
+    this.mentionPrefix = Boolean(options.mentionPrefix)
+
+    this.getGuildPrefix = options.getGuildPrefix ?? ((id: string) => this.prefix)
+    this.getUserPrefix = options.getUserPrefix ?? ((id: string) => this.prefix)
+
+    this.getChannelPrefix = options.getChannelPrefix ?? ((id: string) => this.prefix)
+
+    this.isUserBlacklisted = options.isUserBlacklisted ?? ((id: string) => false)
+    this.isGuildBlacklisted = options.isGuildBlacklisted ?? ((id: string) => false)
+    this.isChannelBlacklisted = options.isChannelBlacklisted ?? ((id: string) => false)
+
+    this.spacesAfterPrefix = options.spacesAfterPrefix ?? false
+
+    this.owners = options.owners ?? []
+    this.allowBots = options.allowBots ?? false
+    this.allowDMs = options.allowDMs ?? true
+    this.caseSensitive = options.caseSensitive ?? false
 
     const self = this as any
     if (self._decoratedCommands !== undefined) {
-      Object.values(self._decoratedCommands).forEach((entry: any) => {
+      (Object.values(self._decoratedCommands)).forEach((entry: any) => {
         this.commands.add(entry)
       })
       self._decoratedCommands = undefined
     }
 
-    this.on(
-      'messageCreate',
-      async (msg: Message) => await this.processMessage(msg)
-    )
+    this.on('messageCreate', async (msg: Message) => await this.processMessage(msg))
```

Comment:
Maybe call it when processing event itself instead of adding listener? (in `src/gateway/handlers/messageCreate.ts`)